packages/main/src/SliderBase.js

```diff
@@ -0,0 +1,648 @@
+		if (this.isPropertyUpdated("min", "max", ...values)) {
+			this._normalizeMinMaxValues(this.min, this.max);
+			this._drawDefaultTickmarks(this.step, this.max, this.min);
+			this.storePropertyState("min", "max");
+
+			// Here the value props are changed programatically (not by user interaction)
+			// and it won't be "stepified" (rounded to the nearest step). 'Clip' them within
+			// min and max bounderies and update the previous state reference.
+			values.forEach(valueType => {
+				const normalizedValue = SliderBase.clipValue(this[valueType], this.min, this.max);
+				this.updateValue(valueType, normalizedValue);
+				this.storePropertyState(valueType);
+			});
+		}
+	}
+
+	/**
+	 * In order to always keep the visual UI representation and the internal
+	 * state in sync, the component has a 'state storage' that is updated when the
+	 * current state is changed due to a user action.
+	 *
+	 * Check if the previously saved state is outdated. That would mean
+	 * a property has been changed programatically because the previous state
+	 * is always updated in the interaction handlers.
+	 *
+	 * Will return true if any of the properties is not equal to its previously
+	 * stored value.
+	 *
+	 * @protected
+	 */
+	isCurrentStateOutdated() {
+		return Object.entries(this._stateStorage).some(([propName, propValue]) => this[propName] !== propValue);
+	}
+
+	/**
+	 * Returns the last stored value of a property
+	 *
+	 * @protected
+	 */
+	getStoredPropertyState(property) {
+		return this._stateStorage[property];
+	}
+
+	/**
+	 * Check if one or more properties have been updated compared to their last
+	 * saved values in the state storage.
+	 *
+	 * @protected
+	 */
+	isPropertyUpdated(...properties) {
+		return properties.some(prop => this.getStoredPropertyState(prop) !== this[prop]);
+	}
+
+	/**
+	 * Updates the previously saved in the _stateStorage values of one or more properties.
+	 *
+	 * @protected
+	 */
+	storePropertyState(...props) {
+		props.forEach(property => {
+			this._stateStorage[property] = this[property];
+		});
+	}
+
+	/**
+	 * Returns the start side of a direction - left for LTR, right for RTL
+	 *
+	 * @protected
+	 */
+	get directionStart() {
+		return this.effectiveDir === "rtl" ? "right" : "left";
+	}
+
+	_normalizeMinMaxValues(min, max) {
+		if (min > max) {
+			this.min = max;
+			this.max = min;
+		}
+	}
+
+	/**
+	 * Calculates and draws the tickmarks with a CSS gradient style
+	 *
+	 * @private
+	 */
+	_drawDefaultTickmarks(step, max, min) {
+		if (!this.showTickmarks || !this.step) {
+			return;
+		}
+
+		// Convert number values to strings to let the CSS do calculations better
+		// rounding/subpixel behavior" and the most precise tickmarks distribution
+		const maxStr = String(max);
+		const minStr = String(min);
+		const stepStr = String(step);
+		const tickmarkWidth = "1px";
+
+		// There is a CSS bug with the 'currentcolor' value of a CSS gradient that does not
```

Comment:
I see. Still, I would prefer to use CSS variables, even from within JS.

```js
getComputedStyle(document.documentElement)
    .getPropertyValue('__ui5_slider_currentcolor_something....')
```

As for IE, we'll have to see how we can achieve the same with the ponyfill. So for now leave it but this should be refactored asap.